app/models/mixins/ownership_mixin.rb

```diff
@@ -11,9 +11,27 @@ module OwnershipMixin
     virtual_column :owned_by_current_user,                :type => :boolean,    :uses => :evm_owner_userid
     virtual_column :owning_ldap_group,                    :type => :string,     :uses => :miq_group
     virtual_column :owned_by_current_ldap_group,          :type => :boolean,    :uses => :owning_ldap_group
+
+    # Determine whether to return vms ownded by the current user or not
+    # Requires `with_miq_group` scope to function
+    #
+    # Resulting SQL: "(LOWER(miq_groups.description) = 'some_miq_group'"
+    #
+    # Will result in the following when used with MiqExpression:
+    #
+    #   WHERE ((LOWER("miq_groups"."description") = 'some_miq_group') = 'true')
+    virtual_attribute :owned_by_current_ldap_group, :boolean, :arel => (lambda do |t|
+      ldap_group = User.current_user.try(:ldap_group).to_s.downcase
+
+      t.grouping(MiqGroup.arel_table[:description].lower.eq(ldap_group))
+    end)
   end
 
   module ClassMethods
+    def with_miq_group
+      select(arel_table[Arel.star]).joins(:miq_group)
```

Comment:
In this case, I am choosing **not** to return any data from the `miq_groups` table, and not from any other tables.  So it actually ends up being the following:

```SQL
SELECT "vms"."*"
```

Or whatever table this model actually happens to be associated with.